frb_dart/lib/src/env/io.dart

```diff
@@ -0,0 +1,33 @@
+// see https://github.com/flutter/flutter/issues/55870
+
+import 'dart:io';
+
+const truthyValues = ['true', '1', 'yes'];
+
+/// ✅ correctly read e.g. export JSON=true && ...
+/// ❌ fail at reading e.g. dart --define=SILICON=true ...
+/// ❌ fail at reading e.g. dart -DSILICON=true ...
+bool? readEnvFromPlatform(String envVarName) =>
+    (Platform.environment[envVarName] != null &&
+            Platform.environment[envVarName]!.isNotEmpty)
+        ? truthyValues.contains(Platform.environment[envVarName]!.toLowerCase())
+        : null;
+
+/// ✅ correctly read e.g. dart --define=SILICON=true ...
+/// ✅ correctly read e.g. dart -DSILICON=true ...
+/// ❌ fail at reading e.g. export JSON=true && ...
+bool? readEnvFromPrimitive(String envVarName) =>
```

Comment:
uber-nit: Does this really work 👀 From https://github.com/flutter/flutter/issues/55870 it needs to be `const bool.fromEnvironment` (and thus cannot be wrapped like this IIRC)

Reply:
well on my machine it works lol
yeah I'm gonna add it you're utterly right anyway :)

Reply:
ah, actually as soon as you turn it into a `const`, then all its origin should be `const` too, which is prevented by `envVarName` here, and even if I inline the function in its parent, the parent method also has parameters. In short it's either I would have to copy paste this long conditional expression around, either make the concession for it to be a  `bool?`.

and actually a `T?` for an env var in general is good, since its nullity informs that it hasn't been set by the user.

Reply:
Interesting - then https://github.com/flutter/flutter/issues/55870 looks outdated

Reply:
hence why I tried to explain as much as possible. on this PR I really think I got all possible bugs one can face lol

Reply:
done in 461e0d3